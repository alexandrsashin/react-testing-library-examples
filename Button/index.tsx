import { ReactNode } from "react";

interface ButtonProps {
  type: "button" | "submit" | "reset";
  children: ReactNode;
  onClick: () => void;
}

const Button = ({ type, children, onClick }: ButtonProps) => {
  return (
    <button type={type} onClick={() => onClick()}>
      {children}
    </button>
  );
};

export default Button;
