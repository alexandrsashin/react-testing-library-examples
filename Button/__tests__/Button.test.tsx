import {
  cleanup,
  screen,
  render,
  RenderResult,
  fireEvent,
} from "@testing-library/react";
import Button from "..";

const onClick = jest.fn();

describe("<Button />", () => {
  let view: RenderResult;

  afterEach(() => {
    cleanup();
    jest.clearAllMocks();
  });

  it("should render with type=button", () => {
    view = render(
      <Button type="button" onClick={onClick}>
        Button
      </Button>
    );

    expect(view.asFragment()).toMatchSnapshot();
  });

  it("should render with type=submit", () => {
    view = render(
      <Button type="submit" onClick={onClick}>
        Button
      </Button>
    );

    expect(view.asFragment()).toMatchSnapshot();
  });

  it("should render with type=reset", () => {
    view = render(
      <Button type="reset" onClick={onClick}>
        Button
      </Button>
    );

    expect(view.asFragment()).toMatchSnapshot();
  });

  it("onClick() is called after click", () => {
    render(
      <Button type="button" onClick={onClick}>
        Button
      </Button>
    );
    const button = screen.getByRole("button", { name: "Button" });

    fireEvent.click(button);
    expect(onClick).toHaveBeenCalled();
  });
});
