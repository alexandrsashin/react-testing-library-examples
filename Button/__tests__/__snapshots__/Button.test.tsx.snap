// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render with type=button 1`] = `
<DocumentFragment>
  <button
    type="button"
  >
    Button
  </button>
</DocumentFragment>
`;

exports[`<Button /> should render with type=reset 1`] = `
<DocumentFragment>
  <button
    type="reset"
  >
    Button
  </button>
</DocumentFragment>
`;

exports[`<Button /> should render with type=submit 1`] = `
<DocumentFragment>
  <button
    type="submit"
  >
    Button
  </button>
</DocumentFragment>
`;
